spec/Candid.md

```diff
@@ -519,20 +519,25 @@ type B = A;  // error: cyclic type definition
 In order to allow splitting interface definitions up into multiple files or share common definitions between multiple interfaces, *import* declarations are provided.
 
 ```
-<def>   ::= ... | import <text>
+<def>   ::= ... | import <text> | import* <text>
 ```
 
-An import refers to another interface file by URL. The semantics is that of textual inclusion, except that definitions from the imported file must not refer to definitions from the importing file.
+There are two forms of import: `import` and `import*`. Both `import` and `import*` refer to another interface file by URL. The type definitions from the imported file are textually included in the importing file. The definitions from the imported file must not refer to definitions from the importing file.
```

Comment:
So a natural question is what happens if either or both services have service arguments? It's not clear what the semantics  should be.

Reply:
An alternative might to add `and`/`or` as in Motoko to take the glb/lub of two service types.

Reply:
(but then you'd have to define the glb/lub, if those even exist for Candid's weird subtyping rules).

Reply:
Good catch! I think we can disallow service arguments in the imported file. By definition, service constructor refers to a Wasm module, while service refers to the running canister. Importing a service constructor would imply some kind of static linking, which we don't support anyway.

I think we cannot define glb/lub, the opt rule makes every `opt T` a bottom type...